internal/controller/resource_manager.go

```diff
@@ -71,8 +71,9 @@ func (r *ResourceManager) WatchK8sResources(ctx context.Context, cancel context.
 	defer cancel()
 	// Here intiatate the watch event
 	watchOptions := metav1.ListOptions{
-		Watch:           true,
-		ResourceVersion: resourceVersion,
+		Watch:               true,
+		ResourceVersion:     resourceVersion,
+		AllowWatchBookmarks: true,
 	}
```

Comment:
Remove this. This should be created only in `newWatcher`.
```suggestion
```

You should only need to pass in `resourceVersion` to `watchEvents` below:
https://github.com/illumio/cloud-operator/pull/296/files#diff-74dcb56206004cdfc489594c5a9ca9d721fd2d652e90dcbc4a5bc4462cd86cb8R86
```
	err = r.watchEvents(ctx, resourceVersion, mutationChan)
```